src/auth/role.guard.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  HttpException,
+  HttpStatus,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { JwtService } from '@nestjs/jwt';
+import { Observable } from 'rxjs';
+import { NOT_AUTHORIZED, NO_ACCESS, ROLE } from '../constants';
+import { ROLE_KEY } from './checkRole.decorator';
+
+@Injectable()
+export class RoleGuard implements CanActivate {
+  constructor(private jwtService: JwtService, private reflector: Reflector) {}
+
+  canActivate(
+    context: ExecutionContext,
+  ): boolean | Promise<boolean> | Observable<boolean> {
+    const req = context.switchToHttp().getRequest();
+    try {
+
+      const requiredRoles = this.reflector.getAllAndOverride<string[]>(ROLE_KEY, [
+        context.getHandler(),
+        context.getClass(),
+      ])
+
+      if (!requiredRoles) {
+        return true
+      }
+      const authHeader = req.headers.authorization;
```

Comment:
jwt parsing again! `JWT guard` -> this one, with taking role out of `req.user` obj